### PR TITLE
disable CI for draft PRs

### PR DESCRIPTION
**Pull request by krisis**
## Description
Disable running of CI for draft PRs

## Motivation and Context
Save CI/CD cycles

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] CI/CD

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Looks like something wrong here @krisis

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: .github/workflows/go-cross.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ concurrency:
 
 jobs:
   build:
+    if: github.event.pull_request.draft == false
```
Comment (harshavardhana) [inline .github/workflows/go-cross.yml:18]: Seems like there is a better way

```
on:
  pull_request:
    types: [review_requested, ready_for_review]
```

- https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-event-pull_request
- https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onevent_nametypes

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by klauspost** [comment]
@krisis @harshavardhana I like having feedback on Draft PRs, so I disagree they are a waste of cycles.

I generally don't consider them ready for review before CI succeeds, so I like having that feedback before marking it ready for review.

**Comment by harshavardhana** [comment]
> @krisis @harshavardhana I like having feedback on Draft PRs, so I disagree they are a waste of cycles.
> 
> I generally don't consider them ready for review before CI succeeds, so I like having that feedback before marking it ready for review.

Yes that is correct as well. What I think @krisis was trying to address is there are PRs that don't even build so why run them such as dependency updates etc.

But we can revert it if @krisis also agrees.

**Comment by krisis** [comment]
@klauspost that's a valid point.  We can revert this change.

**Pull request merged**